content/api/vision/_index.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Vision
+---
+
+Your webcam is how your robot will see markers.
+
+## Searching for markers
+Assuming you have a webcam connected, you can use `r.camera.see()` to take a picture. The software will process the picture and returns a list of the markers it sees, sorted by distance, where the first element of the array is the closest marker to the camera.
```

Comment:
I realise this isn't quite a docs thing, but _why_ are we sorting the markers by distance? This seems completely gratuitous and not good API design.

Reply:
Saves the students having to sort them? :) (Sorry, I don't really understand why it's bad API design)

Reply:
There are a number of reasons I don't like the pre-sorting:
1. Sorting a list can be an expensive operation -- it's always an operation which scales at least with the size of the list (`O(N)`) and often _much_ worse. By doing it inside the API we're forcing all clients to bear that cost, rather than letting them decide whether they need to pay it. Assuming that you know what your consumers want is a great way to build an API which works a little better for the case you're building for and really badly for everything else. I'd rather build an API which is generally useful and works well overall.
2. Sorting the list is _really easy_ to do if the client wants it. Our implementation is (as far as I know) purely for apparent convenience and doesn't offer any improvement over the client doing it later. If we think that the students won't be able to work out how to do it and/or want to make it easier to get to that outcome, then I suggest we include a suitable example in the docs. The precedent for this are the extensive example, how-tos and cookbooks in the main Python docs.
   * Note: if our sort was somehow _much_ more efficient then I'd be more open to the pre-sorting, though would instead encourage a flag or separate API to access it. As it is, we're literally calling the same code that the students could (see the call to `sorted` at https://github.com/sourcebots/robot-api/blob/master/robot/camera.py#L32)
3. I don't think it's a good idea for us to make our APIs too "nice" in ways that real APIs aren't as it won't teach the general skills we're actually trying to teach. It will also make it harder when the competitors encounter a less "nice" API. Of course this _doesn't_ mean that we shouldn't make the API easy to use!

Reply:
I think a good compromise would be to add an optional argument to disable sorting. This keeps the API convenient for entry-level users.

Reply:
I think the argument on computation time is fairly mute. The markers are sorted off a constant value, and with only a handful of markers, the computation and sort time for this is minimal, especially when compared to the computation time of the marker scan. 
I don't see much use in having a flag, as the chances of anyone actually _wanting_ unsorted markers is fairly remove. I suggest we either leave as-is, or have them arbitrarily sorted (however they come out `sb-vision`.

Reply:
I agree the computation time here is probably fairly minimal (since `N` is likely to be < 1000 in all cases). I'm most compelled here by the argument about making our APIs feel as much like real-world (good, well designed) APIs.

While I agree that this isn't strictly a documentation thing, if we document it as being the case then we're stuck with it. If we don't document it, then it's undefined behaviour that we can remove.